include/seqan3/search/algorithm/detail/search_scheme_algorithm.hpp

```diff
@@ -0,0 +1,509 @@
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Christopher Pockrandt <christopher.pockrandt AT fu-berlin.de>
+ * \brief Provides the algorithm to search in an index using search schemes.
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include <range/v3/view/slice.hpp>
+
+#include <seqan3/core/metafunction/transformation_trait_or.hpp>
+#include <seqan3/search/algorithm/detail/search_common.hpp>
+#include <seqan3/search/algorithm/detail/search_scheme_precomputed.hpp>
+#include <seqan3/search/algorithm/detail/search_trivial.hpp>
+#include <seqan3/search/fm_index/concept.hpp>
+
+/*!\addtogroup search
+ * \{
+ */
+
+namespace seqan3::detail
+{
+
+/*!\brief Computes a (non-optimal) search scheme. Currently the generated search scheme represents trivial
+ *        backtracking.
+ * \param[in] min_error Minimum number of errors allowed.
+ * \param[in] max_error Maximum number of errors allowed.
+ */
+inline std::vector<search_dyn> compute_ss(uint8_t const min_error, uint8_t const max_error)
+{
+    // TODO: Replace this at least by the pigeonhole principle or even better by 01*0 schemes.
+    std::vector<search_dyn> scheme{{{1}, {min_error}, {max_error}}};
+    return scheme;
+}
```

Comment:
We will soon have more advanced schemes computed at run-time. I think it's easier to track if we move it into a separate function from the very beginning instead of hard-coding the preliminary algorithm into the search-algorithm.